src/components/rules-creation-page/steps/IncludeExcludeSelectionStep.tsx

```diff
@@ -0,0 +1,236 @@
+
+        console.log(syncRule.metadataExcludeIncludeRules);
+    }, [d2, syncRule]);
+
+    const changeUseDefaultIncludeExclude = (useDefault: boolean) => {
+        onChange(
+            useDefault
+                ? syncRule.markToUseDefaultIncludeExclude()
+                : syncRule.markToNotUseDefaultIncludeExclude(models)
+        );
+    };
+
+    const changeModelName = (event: any) => {
+        setSelectedType(event.target.value);
+    };
+
+    const changeInclude = (includeRules: any) => {
+        const type: string = selectedType || "";
+
+        const oldIncludeRules = getIncludeRules();
+        const oldExcludeRules = getExcludeRules();
+
+        const ruleIndexesToExclude = _.difference(oldIncludeRules, includeRules).map(rule =>
+            oldIncludeRules.indexOf(rule)
+        );
+        const ruleIndexesToInclude = _.difference(includeRules, oldIncludeRules).map(rule =>
+            oldExcludeRules.indexOf(rule)
+        );
+
+        if (ruleIndexesToInclude.length > 0) {
+            onChange(syncRule.moveRuleFromExcludeToInclude(type, ruleIndexesToInclude));
+        } else if (ruleIndexesToExclude.length > 0) {
+            onChange(syncRule.moveRuleFromIncludeToExclude(type, ruleIndexesToExclude));
+        }
+    };
+
+    const getAllRules = () => {
+        const allRules =
+            selectedType && syncRule.metadataExcludeIncludeRules
+                ? [
+                      ...syncRule.metadataExcludeIncludeRules[selectedType].excludeRules,
+                      ...syncRule.metadataExcludeIncludeRules[selectedType].includeRules,
+                  ]
+                : [];
+
+        return allRules;
+    };
+
+    const getIncludeRules = () => {
+        return selectedType && syncRule.metadataExcludeIncludeRules
+            ? syncRule.metadataExcludeIncludeRules[selectedType].includeRules
+            : [];
+    };
+
+    const getExcludeRules = () => {
+        return selectedType && syncRule.metadataExcludeIncludeRules
+            ? syncRule.metadataExcludeIncludeRules[selectedType].excludeRules
+            : [];
+    };
+
+    const getRuleOptions = () => {
+        const allRules = getAllRules();
+
+        return allRules.map(rule => ({
+            value: rule,
+            text: includeExcludeRulesFriendlyNames[rule]
+                ? includeExcludeRulesFriendlyNames[rule]
+                : rule,
+        }));
+    };
+
+    return (
+        <React.Fragment>
+            <Toggle
+                label={i18n.t("Use default configuration")}
+                value={syncRule.useDefaultIncludeExclude}
+                onValueChange={changeUseDefaultIncludeExclude}
+            />
+            {!syncRule.useDefaultIncludeExclude && (
+                <div
+                    style={{
+                        display: "flex",
+                        flexDirection: "column",
+                        alignItems: "flex-start",
+                        marginTop: "20px",
+                    }}
```

Comment:
All the styling should be done through material-ui, sometimes in development looks good with inlined styles but on production when the DOM classes are optimized there're problems.

You can check examples on other components:

```
const useStyles = makeStyles({
    className: {
        // Styling
    },
});
```

Reply:
The development branch has material-ui 3 and makeStyles does not exists in this version.

Reply:
Okay then. we could use the HOC and migrate with malaria to the hook version.

```
export default withStyles(styles)(Component name);
```

Reply:
Done!